libs/utils/include/celix_ref.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CELIX_CELIX_REF_H
+#define CELIX_CELIX_REF_H
+
+#include <stdatomic.h>
+#include <limits.h>
+#include <assert.h>
+
+struct celix_ref {
+    atomic_int count;
+};
+
+static inline void celix_ref_init(struct celix_ref *ref) {
```

Comment:
I would prefer some doxygen documentation explaining the functions from a users perspective

Reply:
CELIX_ and Doxygen added.